autogpt_platform/backend/backend/copilot/expert_context_test.py

```
@@ -105,22 +108,25 @@ async def test_plain_session_returns_empty(self):
+    async def test_archived_expert_hidden_by_accessor_raises(self):
         mock_db = MagicMock()
-        mock_db.get_expert = AsyncMock(return_value=_expert(is_archived=True))
+        mock_db.get_expert = AsyncMock(return_value=None)
         with patch(f"{_EC}.experts_db", MagicMock(return_value=mock_db)):
-            result = await build_expert_identity_suffix("user-1", "exp-1")
-
-        assert result == ""
+            with pytest.raises(
+                ExpertSessionUnavailableError,
+                match="no longer exists or is archived",
+            ):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)